app/controllers/api/v1/meetings_controller.rb

```diff
@@ -124,7 +124,11 @@ def authorized_as_moderator?(mod_code:, viewer_code:, anyone_join_as_mod:)
       end
 
       def authorized_to_start_meeting?(settings, bbb_role)
-        settings['glAnyoneCanStart'] == 'true' || bbb_role == 'Moderator'
+        if settings['glAnyoneJoinAsModerator'] == 'true'
+          settings['glAnyoneCanStart'] == 'true' || @room.user_id == current_user&.id
+        else
+          settings['glAnyoneCanStart'] == 'true' || bbb_role == 'Moderator'
+        end
```

Comment:
Can we not just do only `settings['glAnyoneCanStart'] == 'true' || @room.user_id == current_user&.id || room is shared`

Reply:
It's a little more complicated actually

Reply:
`settings['glAnyoneCanStart'] == 'true' || @room.user_id == current_user&.id || current_user&.shared_rooms&.include?(@room)`

Reply:
updated as you mentioned!